runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionContext.java

```diff
@@ -307,5 +335,34 @@ public Closeable enterState(ExecutionState newState) {
     public String getWorkItemId() {
       return this.workItemId;
     }
+
+    public ActiveMessageMetadata getActiveMessageMetadata() {
```

Comment:
If this is nullable, recommended to make it explicit by returning `Optional<ActiveMessageMetadata>` [go/java-practices/optional]

Reply:
ef56c0419a649d246e0ee00e927ee821361be943